MapboxNavigationUI/MGLMapView.swift

```diff
@@ -145,31 +153,21 @@ extension MGLMapView {
             arrowStroke.lineWidth = MGLStyleValue(rawValue: 8)
             arrowStroke.lineColor = MGLStyleValue(rawValue: NavigationUI.shared.tintColor)
             
-            style?.addSource(arrowSourceStroke)
-            style?.addSource(arrowSource)
+            if let source = style.source(withIdentifier: arrowSourceIdentifier) {
+                let s = source as! MGLShapeSource
+                s.shape = MGLShapeCollection(shapes: maneuverArrowPolylines)
+            } else {
+                style.addSource(arrowSource)
+                style.addLayer(arrow)
+            }
             
-            style?.addLayer(arrow)
-            style?.insertLayer(arrowStroke, below: arrow)
-        }
-    }
-    
-    func removeArrow() {
-        guard let style = style else { return }
-        
-        if let arrow = style.layer(withIdentifier: "arrow") {
-            style.removeLayer(arrow)
-        }
-        
-        if let arrow = style.layer(withIdentifier: "arrowStroke") {
-            style.removeLayer(arrow)
-        }
-        
-        if let arrowStrokeSourceCheck = style.source(withIdentifier: "arrowSourceStroke") {
-            style.removeSource(arrowStrokeSourceCheck)
-        }
-        
-        if let arrowSourceCheck = style.source(withIdentifier: "arrowSource") {
-            style.removeSource(arrowSourceCheck)
+            if let source = style.source(withIdentifier: arrowSourceStrokeIdentifier) {
+                let s = source as! MGLShapeSource
+                s.shape = MGLShapeCollection(shapes: maneuverArrowStrokePolylines)
+            } else {
+                style.addSource(arrowSourceStroke)
+                style.insertLayer(arrowStroke, below: arrow)
```

Comment:
Likewise, move all the code that sets up `arrowSourceStroke` and `arrowStroke` into this `else` block.